src/library_async.js

```diff
@@ -44,6 +44,13 @@ mergeInto(LibraryManager.library, {
           var original = imports[x];
           var sig = original.sig;
           if (typeof original == 'function') {
+#if ASYNCIFY == 2
+            if (!!x.match('^invoke_[dfijv][dfij]*$')) {
+              // Add the suspender (a pointer) as the first parameter, in line with
+              // suspending: 'first' that we add to the WebAssembly.Function that we create below.
+              original.sig = sig = x.substr(7, 1) + 'i' + x.substr(8);
+            }
```

Comment:
Would we not instead add the `.sig` when we create the invoke functions?

Do you have an example/test that shows when this is needed?

Reply:
As for a suitable example:

`#include <stdio.h>
#include <emscripten.h>
#include <emscripten/html5.h>

EM_JS(void, suspend, (), {
  return Asyncify.handleSleep(function(wakeUp) {
    Module.resume = wakeUp;
  });
});

EM_JS(void, resume, (), {
  Module.resume();
});

int main() {
  emscripten_set_timeout(resume, 100, (void*)0);
  printf("suspend\n");
  suspend();
  printf("finish\n");
}`

Reply:
I pushed a new change with the above problem handled and with signature assigned on invoke_* creation. See if you like it :)